Tests/JetworkingTests/Mocks/MockExecuter.swift

```diff
@@ -0,0 +1,106 @@
+import Foundation
+@testable import Jetworking
+
+struct MockCancellable: CancellableRequest {
+    var identifier: Int = Int.random(in: (Int.min..<Int.max))
+    var onCancelCalled: ((Int) -> Void)?
+
+    init(onCancelCalled: ((Int) -> Void)?) {
+        self.onCancelCalled = onCancelCalled
+    }
+
+    func cancel() {
+        onCancelCalled?(identifier)
+    }
+}
+
+final class MockExecuter: RequestExecutor {
+    let session: URLSession
+    let encoder: JSONEncoder = JSONEncoder()
+    let defaultCompletionDelay: TimeInterval = 0.5
+    let responseCode: Int = 200
+    let headerFields: [String: String]? = nil
+    var isCancelled: Bool = false
+
+    static var completionDelayForRequest: ((URLRequest) -> TimeInterval)?
+
+    init(session: URLSession) {
+        self.session = session
+    }
+
+    func send(
+        request: URLRequest,
+        _ completion: @escaping ((Data?, URLResponse?, Error?) -> Void)
+    ) -> CancellableRequest? {
+        let completionDelay = MockExecuter.completionDelayForRequest?(request) ?? defaultCompletionDelay
+        DispatchQueue.main.asyncAfter(deadline: .now() + completionDelay) { [weak self] in
+            guard self?.isCancelled == false  else { return completion(nil, nil, URLError(.cancelled)) }
```

Comment:
I would prefer following format:
```
guard
    let self = self,
    self.isCancelled == false
else {
    completion(nil, nil, URLError(.cancelled))
    return
}

self.execute(
    request: request,
    completion: completion
)
```

or at least guarding `self` as you use it below.